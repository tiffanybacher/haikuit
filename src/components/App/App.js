import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import PropTypes from 'prop-types';
import Header from '../Header/Header';
import Menu from '../Menu/Menu';
import HomeContainer from '../HomeContainer/HomeContainer';
import About from '../About/About';
import HaikusContainer from '../HaikusContainer/HaikusContainer';
import EditHaiku from '../EditHaiku/EditHaiku';
import SavedHaiku from '../SavedHaiku/SavedHaiku';
import Page404 from '../Page404/Page404';

class App extends Component {
  state = {
    menuShown: false
  }

  toggleMenu = () => {
    this.setState({
      menuShown: !this.state.menuShown
    });
  }

  renderMenu = () => {
    return this.state.menuShown 
      ? <Menu toggleMenu={this.toggleMenu} /> 
      : null
  }

  render() {
    return (
        <div className="App">
          {this.renderMenu()}
          <Header 
            toggleMenu={this.toggleMenu} 
            menuShown={this.state.menuShown} />
          <Switch>
          <Route exact path="/" component={HomeContainer} />
          <Route exact path="/about" component={About} />
          <Route exact path="/haikus" component={HaikusContainer} />
          <Route exact path="/haiku/:id/edit" component={EditHaiku} />
          <Route exact path="/haiku/:id/saved" component={SavedHaiku} />
          <Route component={Page404} />
          </Switch>
        </div>
    );
  }
}

export default App;

App.propTypes = {
  menuShown: PropTypes.bool
}
