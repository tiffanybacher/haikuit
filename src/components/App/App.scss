body {
  background-color: #EEE;
  font-family: "Raleway", -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
   margin: 0;

  * {
    margin: 0;
    padding: 0;
  }

  button {
    background: none;
    border: none;
    cursor: pointer;
  }

  .flex-container {
    display: flex;
  }
}

.App {
  min-width: 320px;
}