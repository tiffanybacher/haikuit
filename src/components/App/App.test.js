import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state for MenuShown', () => {
    expect(wrapper.state('menuShown')).toBe(false);

    wrapper.instance().toggleMenu();

    expect(wrapper.state('menuShown')).toBe(true);
  });

  it('should match snapshot if menuShown is true', () => {
    wrapper.setState({ menuShown: true });

    expect(wrapper).toMatchSnapshot();
  });
});

