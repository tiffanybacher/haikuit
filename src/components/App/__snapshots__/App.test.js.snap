// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div
  className="App"
>
  <Header
    toggleMenu={[Function]}
  />
</div>
`;

exports[`App should match snapshot if menu is shown 1`] = `
<div
  className="App"
>
  <Menu
    toggleMenu={[Function]}
  />
  <Header
    toggleMenu={[Function]}
  />
</div>
`;
